tests/testSpherePoint.py

```diff
@@ -123,16 +123,21 @@ def testInit1ArgErrors(self):
         SpherePoint(afwGeom.Point3D(0.0, 0.0, 1.0))
         SpherePoint(afwGeom.Point3D(42.78, -46.29, 38.27))
 
+    def testCopyConstructor(self):
+        sp = SpherePoint(-42.0*degrees, 45.0*degrees)
+        spcopy = SpherePoint(sp)
+        self.assertEqual(sp, spcopy)
+
     def testInitNArgFail(self):
         """Tests if only 1- or 2-argument initializers are allowed.
         """
-        with self.assertRaises(NotImplementedError):
+        with self.assertRaises(TypeError):
             SpherePoint()
-        with self.assertRaises(NotImplementedError):
+        with self.assertRaises(TypeError):
             SpherePoint("Rotund", "Bovine")
-        with self.assertRaises(NotImplementedError):
+        with self.assertRaises(TypeError):
             SpherePoint(42)
-        with self.assertRaises(NotImplementedError):
+        with self.assertRaises(TypeError):
             SpherePoint("ICRS", 34.0, -56.0)
```

Comment:
Hmm, this brings to mind we should be careful when we use `TypeError` and where a check for `NotImplemented` is more appropriate. Keep this change here for now, but we will need to reconsider.